microceph/api/pool.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/canonical/lxd/shared/logger"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+)
+
+// /1.0/pool endpoint.
+var poolsCmd = rest.Endpoint{
+	Path: "pools",
```

Comment:
Nit: May I suggest to rename this endpoint to `pools-op/set-rf` for operations on pools (such as setting replication factor). This is so that `pools` endpoint could be reserved for CRUD operations on ceph pools (a possibility for future feature work ?).

Reply:
Also, RESTyness 😄

Reply:
Updated to pools-op.